project/src/test/java/com/google/sps/EventDetailTest.java

```diff
@@ -103,6 +107,34 @@ public void verifyGetAllEvents() throws IOException {
         CommonUtils.convertToJson(expectedEvents).trim(), stringWriter.toString().trim());
   }
 
```

Comment:
I think you should create a separate test for the FilteredEventServlet class since it seems like separate functionality from the EventDetail class.

Reply:
@liangqigong  @hlz03 My reasoning here for using the same test file is that I'm using the exact same constants and both FilteredEventServlet and EventDetailServlet are called on the event-discovery.html page, but should I seperate by servlet instead of by which HTML page will eventually call it? What is the best practice in this case?

Reply:
I think you should separate the test files by individual servlet instead of by which HTML page will eventually call it since the HTML page call can change in the future and you don't want to update your backend test every time your frontend code changes. 

And you can create a test helper class to store all the constants if you want.

Reply:
That makes sense, thanks! Done